### PR TITLE
Feature/v3.x.x/876 torrent client failure handling

### PR DESCRIPTION
**Pull request by Dzejkop**

### CODE REVIEW

File: Assets/PatchKit Patcher/Scripts/AppData/Remote/Downloaders/TorrentDownloader.cs [hunk 1]
```diff
@@ -67,47 +105,25 @@ public void Download(CancellationToken cancellationToken)
...
-                    torrentClient.AddTorrent(_torrentFilePath, DestinationDirectoryPath, cancellationToken);
-
-                    var timeoutWatch = new Stopwatch();
-                    timeoutWatch.Start();
-
-                    TorrentStatus status = GetAndCheckTorrentStatus(torrentClient, cancellationToken);
-                    double initialProgress = status.Progress;
-                    _logger.LogTrace("initialProgress = " + status.Progress);
-                    var waitHandle = new AutoResetEvent(false);
+                IRequestRetryStrategy retryStrategy = new RetryStrategy(3);
```
Comment (witcher112) [inline Assets/PatchKit Patcher/Scripts/AppData/Remote/Downloaders/TorrentDownloader.cs]: Why only three retries?

Comment (Dzejkop) [reply Assets/PatchKit Patcher/Scripts/AppData/Remote/Downloaders/TorrentDownloader.cs]: I picked this value completely arbitrarily

File: Assets/PatchKit Patcher/Scripts/AppData/Remote/Downloaders/TorrentDownloader.cs [hunk 2]
```diff
@@ -67,47 +105,25 @@ public void Download(CancellationToken cancellationToken)
...
-                            if (status.IsSeeding)
-                            {
-                                finished = true;
-                            }
-                            else
-                            {
-                                waitHandle.WaitOne(UpdateInterval);
-                            }
-                        } while (!finished);
+                            throw new DownloadFailureException("Failed to download with torrent-client.");
```
Comment (witcher112) [inline Assets/PatchKit Patcher/Scripts/AppData/Remote/Downloaders/TorrentDownloader.cs]: To keep the error information we should put `e` as inner exception inside `DownloadFailureException`.

File: Assets/PatchKit Patcher/Scripts/AppData/Remote/Downloaders/TorrentClient.cs [hunk 3]
```diff
@@ -101,10 +101,20 @@ public TorrentClientStatus GetStatus(CancellationToken cancellationToken)
             }
         }
 
+        private void ThrowIfProcessExited()
+        {
+            if (_process.HasExited)
+            {
+                throw new TorrentClientException(string.Format("torrent-client has exited with error code {0}", _process.ExitCode));
```
Comment (witcher112) [inline Assets/PatchKit Patcher/Scripts/AppData/Remote/Downloaders/TorrentClient.cs]: The name should be more specific - `TorrentClientCrashException` or something like that.

File: Assets/PatchKit Patcher/Scripts/AppData/Remote/Downloaders/TorrentDownloader.cs (hunk 1 above)
Comment (witcher112) [reply Assets/PatchKit Patcher/Scripts/AppData/Remote/Downloaders/TorrentDownloader.cs]: Ok, it should be fine.

**Pull request merged**